include/mxnet/lib_api.h

```diff
@@ -393,13 +453,21 @@ class OpResource {
     return static_cast<mx_stream_t>(cuda_stream);
   }
 
+  void alloc_sparse(MXSparse* sparse, int index, int indices_len, int indptr_len = 0) {
+    sparse_malloc(sparse_alloc, index, indices_len, indptr_len, 
+                   &(sparse->data), &(sparse->indices), &(sparse->indptr));
+  }
+
  private:
   /*! \brief allocation lambda function */
   xpu_malloc_t cpu_malloc, gpu_malloc;
   /*! \brief lambda function to return allocated memory handle */
   void *cpu_alloc, *gpu_alloc;
   /*! \brief cuda stream passed from MXNet */
   void *cuda_stream;
+
+  sparse_malloc_t sparse_malloc;
+  void *sparse_alloc;
```

Comment:
`/*! \brief lambda function to return allocated sparse memory handle */`

Reply:
Added.